src/api/routes/tx.ts

```diff
@@ -113,6 +113,12 @@ export const TxRoutes: FastifyPluginAsync<
               examples: [1706745599],
             })
           ),
+          search_term: Type.Optional(
+            Type.String({
+              description: 'Option to search for transactions by a search term',
+              examples: ['swap'],
+            })
+          ),
```

Comment:
The idea is to have the `search_term` as a broad search filter that looks into multiple fields. My plan so far is to incrementally add more fields to the same index and test performance, e.g.:
```
CREATE INDEX idx_search_term_trgm
ON public.txs
USING gin (
  contract_call_function_name gin_trgm_ops,
  contract_name gin_trgm_ops
  memo gin_trgm_ops
  ....
);
```
Open to suggestions ofc.

Reply:
That seems fine to me. In the future we could always add another query param that lets users specify a given field(s) to search. And for now it can default to "all".